options/m_option.h

```diff
@@ -424,63 +424,63 @@ struct m_option {
-#define UPDATE_SUB_EXTS         (1 << 23) // update internal list of sub exts
-#define UPDATE_VIDEO            (1 << 24) // force redraw if needed
-#define UPDATE_VO               (1 << 25) // reinit the VO
-#define UPDATE_CLIPBOARD        (1 << 26) // reinit the clipboard
-#define UPDATE_OPT_LAST         (1 << 26)
+#define UPDATE_OPT_FIRST        (UINT64_C(1) << 8)
+#define UPDATE_TERM             (UINT64_C(1) << 8)  // terminal options
+#define UPDATE_SUB_FILT         (UINT64_C(1) << 9)  // subtitle filter options
+#define UPDATE_OSD              (UINT64_C(1) << 10) // related to OSD rendering
+#define UPDATE_BUILTIN_SCRIPTS  (UINT64_C(1) << 11) // osc/ytdl/stats
+#define UPDATE_IMGPAR           (UINT64_C(1) << 12) // video image params overrides
+#define UPDATE_INPUT            (UINT64_C(1) << 13) // mostly --input-* options
+#define UPDATE_AUDIO            (UINT64_C(1) << 14) // --audio-channels etc.
+#define UPDATE_PRIORITY         (UINT64_C(1) << 15) // --priority (Windows-only)
+#define UPDATE_SCREENSAVER      (UINT64_C(1) << 16) // --stop-screensaver
+#define UPDATE_VOL              (UINT64_C(1) << 17) // softvol related options
+#define UPDATE_LAVFI_COMPLEX    (UINT64_C(1) << 18) // --lavfi-complex
+#define UPDATE_HWDEC            (UINT64_C(1) << 20) // --hwdec
+#define UPDATE_DVB_PROG         (UINT64_C(1) << 21) // some --dvbin-...
+#define UPDATE_SUB_HARD         (UINT64_C(1) << 22) // subtitle opts. that need full reinit
+#define UPDATE_SUB_EXTS         (UINT64_C(1) << 23) // update internal list of sub exts
+#define UPDATE_VIDEO            (UINT64_C(1) << 24) // force redraw if needed
+#define UPDATE_VO               (UINT64_C(1) << 25) // reinit the VO
+#define UPDATE_CLIPBOARD        (UINT64_C(1) << 26) // reinit the clipboard
+#define UPDATE_OPT_LAST         (UINT64_C(1) << 26)
 
 // All bits between _FIRST and _LAST (inclusive)
 #define UPDATE_OPTS_MASK \
-    (((UPDATE_OPT_LAST << 1) - 1) & ~(unsigned)(UPDATE_OPT_FIRST - 1))
+    (((UPDATE_OPT_LAST << 1) - 1) & ~(UPDATE_OPT_FIRST - 1))
 
 // type_float/type_double: string "default" is parsed as NaN (and reverse)
-#define M_OPT_DEFAULT_NAN       (1 << 26)
+#define M_OPT_DEFAULT_NAN       (UINT64_C(1) << 59)
 
 // type time: string "no" maps to MP_NOPTS_VALUE (if unset, NOPTS is rejected)
 // and
 // parsing: "--no-opt" is parsed as "--opt=no"
-#define M_OPT_ALLOW_NO          (1 << 27)
+#define M_OPT_ALLOW_NO          (UINT64_C(1) << 60)
 
 // type channels: disallow "auto" (still accept ""), limit list to at most 1 item.
-#define M_OPT_CHANNELS_LIMITED  (1 << 28)
+#define M_OPT_CHANNELS_LIMITED  (UINT64_C(1) << 61)
 
 // type_float/type_double: controls if pretty print should trim trailing zeros
-#define M_OPT_FIXED_LEN_PRINT   (1 << 29)
+#define M_OPT_FIXED_LEN_PRINT   (UINT64_C(1) << 62)
 
 // Like M_OPT_TYPE_OPTIONAL_PARAM.
-#define M_OPT_OPTIONAL_PARAM    (1 << 30)
+#define M_OPT_OPTIONAL_PARAM    (UINT64_C(1) << 63)
```

Comment:
Could we start from 63 and count down? So we can add new flags here instead of before NAN one.

Reply:
And after last flag we can add `static_cast(UPDATE_OPT_LAST < M_OPT_OPTIONAL_PARAM)`, so it is easily visible requirement.